### DISCUSSION TITLE
Higher resolution images

### DISCUSSION THREAD

**Discussion by matty0005** [damongolding/immich-kiosk#19] [Ideas]
Are we able to have an option to use higher resolution images? 

At the moment, the images look a bit blurry and pixelated (26MP images which look crisp normally on immich).

**Comment by damongolding** [comment]
I'll have a look into it. I believe I did have the higher resolution images being pulled in 0.1.0 but I can't remember why I removed that. Possibly thought it was overkill but it would be nice to give that option to users.

**Comment by damongolding** [comment]
Do you have an example @matty0005?  Kiosk pulls from the same API endpoint as Immich's front end and when I compare the same image in both Immich and Kiosk it's the same image data wise.

**Comment by damongolding** [comment]
![Screenshot 2024-07-28 at 8 18 54 am](https://github.com/user-attachments/assets/77046c95-49a3-4cc4-ae56-73dd3c3f0d71)

The above settings in Immich `settings>image settings` could be the answer you are looking for @matty0005

**Comment by SelfSufficient08** [comment]
@matty0005 - I think what you are asking for is for the ability to set a minimum resolution so that all photos are at or above that resolution.  I think I see what you see on my tablets where it is either pulling in the thumbnail versions of photos and the high res ones.  OR maybe at some point during imports, I uploaded a low res and a high res photo.  I need to spend time digging into the library to try and figure out why I see so many low res photos coming up on my devices...  I might need to edit the thumbnail setting above and reprocess the library to see what it does.  I just have 76,000+ photos in my library. lol.

**Reply by SelfSufficient08** [reply]
I can confirm that the issue is more on the immich side.  Immich includes thumbnails it generates in searches.  I am adjusting my thumbnail size to 720p and then regenerating them to see if this fixes it.  The only potential fix for Kiosk to resolve would be to have a minimum resolution setting.

**Reply by matty0005** [reply]
Not so much about selecting only the higher resolution ones - all my photos in immich have a sensible resolution, its just that its as if the full resolution copy of the image isn't been loaded, and rather some sort of thumbnail image is been used.

**Comment by damongolding** [comment]
I’m not sure that is the issue (but I can’t rule it out) as I specifically request the “preview” size, which is what you see when you click on a thumbnail in Immich.

I’d suggest upping the preview size to 4K for the maximum size Immich will output (apart from the original). Then I personally use WebP for smaller file sizes.

I did try getting the original image but the server does take a large performance hit, especially when blurring an image to use for the background.

**Reply by matty0005** [reply]
Ahh yep that would be it. I do notice when i click on the image in immich it doesnt load the highest res copy. After i zoom in, it will then load the higher res copy a second or two later. I will update the preview size.

In the meantime are we able to have an env variable to allow us to use the full res copy even if we are ok with taking the performance hit?

**Reply by damongolding** [reply]
After reading this I saw Immich loads in the original once zoom mode was enabled (by clicking the zoom button). 

As it stands I'm not actively developing Kiosk at the moment. So I won't be adding anything to Kiosk for a little while. You can read the reason on my announcement post in discussions.

**Comment by SelfSufficient08** [comment]
I fortunately have a person that doesnt have many photos tied to them.  I did a search on that person and scrolled to through the photos in immich and found that every video file has a thumbnail photo listed in the library as if it were a real photo.  I determined it was the thumbnail because all 70 of them in this persons case is 333 x 250.  Which aligns with the thumbnail setting.  So this issue might be very specific to videos that immich generates thumbnails for.  Since my wife and I have cell phones, most of our photos have video snippets with them.  My library is rather large so it will probably be a few days until I can see what it does.  I also may need to manually cleanup the library and fix broken connections according to some forum posts in the immich world.  

<img width="979" alt="Screenshot 2024-08-02 at 9 41 26 AM" src="https://github.com/user-attachments/assets/4446a3b0-e6a1-4151-bf5a-0d03e19d8afc">

**Comment by damongolding** [comment]
And said thumbnail is appearing in Kiosk? It’s something I’ll have to look into once I’m back if that’s the case.

**Reply by SelfSufficient08** [reply]
Yes, it appears they do.  It makes sense that they do because they show in immich when I select a filter on a person.  
<img width="618" alt="Screenshot 2024-08-02 at 11 04 16 AM" src="https://github.com/user-attachments/assets/4a8b3619-7f20-44b3-be5f-8daaed60a97d">

**Comment by damongolding** [comment]
My immediate thought would be to filter out filenames that have “-thumbnail” in them. Or see if there is a way for me to see what the users thumbnail settings are and match images against those, and then filter them out i.e. webp format, 250p etc.

**Reply by SelfSufficient08** [reply]
I think the filter our -thumbnail is a perfect solution. At least based on what I am seeing.

**Reply by SelfSufficient08** [reply]
> My immediate thought would be to filter out filenames that have “-thumbnail” in them. Or see if there is a way for me to see what the users thumbnail settings are and match images against those, and then filter them out i.e. webp format, 250p etc.

@damongolding  - Do you think this is something you could implement ?  I still see a ton of fuzzy photos because of the thumbnail generation on live photos.   Thanks again for a great app!

**Reply by damongolding** [reply]
I apologise that this fell off my radar. I will ask the Immich devs if Immich generates these files and if so what the naming convention is. Then I can decide if it's better to exclude them by default or have a config option.

**Reply by damongolding** [reply]
@SelfSufficient08 I have been chatting to a Immich dev. Are you able to sure your Immich compose and env file with me? We possible think you may have mounted the upload folder as an external library. This would result in file recursion cause Immich to use preview/thumbnail files as actual image files.

**Reply by SelfSufficient08** [reply]
Yes sir !!  No need to be sorry !  I love this app and I am sorry to be a pain.  Compose and Env files below.

**### Compose File:**

services:
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
      - ${EXTERNAL_LIBRARY}:/mnt/media/zdrive:ro
    networks:
      - immich
      - proxy
    env_file:
      - .env
    ports:
      - 2283:2283
    depends_on:
      - redis
      - database
    restart: always
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.photos.entrypoints=http"
      - "traefik.http.routers.photos.rule=Host(`photos.photo.com`)"
      - "traefik.http.middlewares.photos-https-redirect.redirectscheme.scheme=https"
      - "traefik.http.routers.photos.middlewares=photos-https-redirect"
      - "traefik.http.routers.photos-secure.entrypoints=https"
      - "traefik.http.routers.photos-secure.rule=Host(`photos.photo.com`)"
      - "traefik.http.routers.photos-secure.tls=true"
      - "traefik.http.routers.photos-secure.service=photos"
      - "traefik.http.services.photos.loadbalancer.server.port=3001"
      - "traefik.docker.network=proxy"


  immich-microservices:
    container_name: immich_microservices
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
      - ${EXTERNAL_LIBRARY}:/mnt/media/zdrive:ro
    networks:
      - immich
    env_file:
      - .env
    depends_on:
      - redis
      - database
    restart: always

  immich-machine-learning:
    container_name: immich_machine_learning
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}
    volumes:
      - model-cache:/cache
    networks:
      - immich
    env_file:
      - .env
    restart: always

  redis:
    container_name: immich_redis
    image: registry.hub.docker.com/library/redis:6.2-alpine@sha256:84882e87b54734154586e5f8abd4dce69fe7311315e2fc6d67c29614c8de2672
    restart: always
    networks:
      - immich

  database:
    container_name: immich_postgres
    image: registry.hub.docker.com/tensorchord/pgvecto-rs:pg14-v0.2.0@sha256:90724186f0a3517cf6914295b5ab410db9ce23190a2d9d0b9dd6463e3fa298f0
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
    volumes:
      - ${DB_DATA_LOCATION}:/var/lib/postgresql/data
    networks:
      - immich
    restart: always

volumes:
  model-cache:


networks:
  proxy:
    external: true
  immich:
    name: immich
`
**### ENV File:** 

UPLOAD_LOCATION=/ZFS-16TB/data/Z_Drive/Pictures-Video/immich-library
EXTERNAL_LIBRARY=/ZFS-16TB/data/Z_Drive/Pictures-Video

DB_DATA_LOCATION=./postgres

IMMICH_VERSION=release

DB_PASSWORD=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX


DB_USERNAME=postgres
DB_DATABASE_NAME=immich`

**Reply by damongolding** [reply]
Ok I see the issue. You have your upload location inside an external library. So when you upload an image, Immich will generate the thumbs which are being picked up by the external library. Immich is then importing the thumbnails via the external library.

```sh
UPLOAD_LOCATION=/ZFS-16TB/data/Z_Drive/Pictures-Video/immich-library
EXTERNAL_LIBRARY=/ZFS-16TB/data/Z_Drive/Pictures-Video
```

Does that make sense?

**Comment by SelfSufficient08** [comment]
I do follow.  I need to move it a layer above to avoid the library from
rescanning it and making them part of the library itself.  I am sorry that
I wasted your time with this.  I appreciate your help !

**Comment by damongolding** [comment]
No need to apologise! I'm glad we got to the bottom of the issue 😊

**Reply by SelfSufficient08** [reply]
I moved the folder higher in the directory so that it's not scanned, did a search for anything in the library folder that included -thumbnail and deleted it from the main library. Now I think I just need to wait a day for the system to clean itself.
